kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt

```diff
@@ -0,0 +1,296 @@
+   */
+  private fun performTrace(
+    utilityNetwork: UtilityNetwork,
+    traceConfiguration: UtilityTraceConfiguration,
+    startingLocation: UtilityElement
+  ) {
+    progressBar.visibility = View.VISIBLE
+    // create a category comparison for the trace
+    // NOTE: UtilityNetworkAttributeComparison or UtilityCategoryComparison with Operator.DoesNotExists
+    // can also be used. These conditions can be joined with either UtilityTraceOrCondition or UtilityTraceAndCondition.
+    val categoryComparison = UtilityCategoryComparison(
+      spinner.selectedItem as UtilityCategory,
+      UtilityCategoryComparisonOperator.EXISTS
+    )
+    // set the category comparison to the barriers of the configuration's trace filter
+    traceConfiguration.filter.barriers = categoryComparison
+
+    // set the configuration to include or leave out isolated features
+    traceConfiguration.isIncludeIsolatedFeatures = include_isolated_switch.isChecked
+
+    // build parameters for the isolation trace
+    val traceParameters =
+      UtilityTraceParameters(UtilityTraceType.ISOLATION, listOf(startingLocation))
+    traceParameters.traceConfiguration = traceConfiguration
+
+    // run the trace and get the result
+    val utilityTraceResultsFuture = utilityNetwork.traceAsync(traceParameters);
+    utilityTraceResultsFuture.addDoneListener {
+      try {
+        (utilityTraceResultsFuture.get()[0] as? UtilityElementTraceResult)?.let { utilityElementTraceResult ->
+          if (utilityElementTraceResult.elements.isNotEmpty()) {
+            Log.i(
+              TAG,
+              "I think the trace was successful ${utilityElementTraceResult.elements.size}"
+            )
+            // clear any selections from a previous trace
+            mapView.map.operationalLayers.filterIsInstance<FeatureLayer>()
+              .forEach { featureLayer ->
+                featureLayer.clearSelection()
+
+                with(QueryParameters()) {
+                  utilityElementTraceResult.elements.filter { it.networkSource.name == featureLayer.featureTable.tableName }
+                    .forEach { utilityElement ->
+                      this.objectIds.add(utilityElement.objectId)
+                    }
+
+                  // select features that match the query
+                  featureLayer.selectFeaturesAsync(this, FeatureLayer.SelectionMode.NEW)
+                }
+              }
+          } else {
+            // trace result is empty
+            Toast.makeText(
+              this,
+              "Utility Element Trace Result had no elements!",
+              Toast.LENGTH_LONG
+            ).show()
+          }
+        }
+        // hide the progress bar when the trace is completed or failed
+        progressBar.visibility = View.GONE
+      } catch (e: Exception) {
+        val message = "Error loading utility trace results: ${e.message}"
+        Log.e(TAG, message)
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
```

Comment:
```suggestion
        val error = "Error loading utility trace results: ${e.message}"
        Log.e(TAG, error)
        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
```

Reply:
Quite nitpicky, but if it's going out to Log.e it's probably more accurately described as an error Log.i might fit message better